modules/display.c

```diff
@@ -365,6 +367,39 @@ static gboolean backlight_ioctl(int value)
 	return status;
 }
 
+/* Set DPMS mode for X11 display to on or off
+ *
+ * @params dpms_on To force the mode into on or off (blanked) mode.
+ */
+static void dpms_set_display_on(gboolean dpms_on)
+{
+	Display* dpy = NULL;
+
+	dpy = XOpenDisplay(NULL);
+	if (dpy == NULL) {
+		dpy = XOpenDisplay(":0.0");
+	}
+
+	if (dpy == NULL) {
+		mce_log(LL_INFO, "%s: unable to open display", __func__);
+		return;
+	}
+
+	if (DPMSCapable(dpy)) {
+		if (dpms_on) {
+			DPMSEnable(dpy);
+			DPMSForceLevel(dpy, DPMSModeOn);
+		} else {
+			DPMSEnable(dpy);
+			usleep(100000);
+			DPMSForceLevel(dpy, DPMSModeOff);
+		}
```

Comment:
We have the same expression in both branches: `DPMSEnable(dpy);`
It's better to move it in common place.
Something like this:
```
DPMSEnable(dpy);
if (dpms_on) {
	DPMSForceLevel(dpy, DPMSModeOn);
} else {
	usleep(100000);
	DPMSForceLevel(dpy, DPMSModeOff);
}
```

Reply:
Why do we need to call DPMSEnable(dpy); every time? Isn't it better to call it once and keep the state in a variable?